File: source/aoBlogs2/Views/BlogImageView.vb

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Text.RegularExpressions
Imports Contensive.Addons.Blog.Controllers
Imports Contensive.Addons.Blog.Models
Imports Contensive.BaseClasses

Namespace Views
    '
    Public Class BlogImageView
        '
        '========================================================================
        '
        Public Shared Function getBlogImage(cp As CPBaseClass, app As ApplicationEnvironmentModel, blogImage As BlogImageModel, ByRef Return_ThumbnailFilename As String, ByRef Return_ImageFilename As String, ByRef Return_ImageDescription As String, ByRef Return_Imagename As String) As String
            Dim results As String = ""
            Try
                '
                If (blogImage IsNot Nothing) Then
                    Dim imageSizeList As List(Of String) = blogImage.AltSizeList.Split(","c).ToList()
                    Return_ImageDescription = blogImage.description
                    Return_Imagename = blogImage.name
                    Return_ThumbnailFilename = cp.Image.GetBestFitWebP(blogImage.Filename, app.blog.ImageWidthMax, CInt(app.blog.ImageWidthMax * 0.75), imageSizeList)
                    Return_ImageFilename = cp.Image.GetBestFitWebP(blogImage.Filename, app.blog.ThumbnailImageWidth, 0, imageSizeList)
                    If (String.Join(",", imageSizeList) <> blogImage.AltSizeList) Then
                        blogImage.AltSizeList = String.Join(",", imageSizeList)
                        blogImage.save(Of BlogImageModel)(cp)
                    End If
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return results
        End Function
        '
        '==================================================================================
        '
        '
    End Class
End Namespace

File: source/aoBlogs2/Views/BlogImageView.cs

using System;
using System.Linq;
using Contensive.Blog.Models;
using Contensive.BaseClasses;

namespace Contensive.Blog.Views {
    // 
    public class BlogImageView {
        // 
        // ========================================================================
        // 
        public static string getBlogImage(CPBaseClass cp, ApplicationEnvironmentModel app, BlogImageModel blogImage, ref string Return_ThumbnailFilename, ref string Return_ImageFilename, ref string Return_ImageDescription, ref string Return_Imagename) {
            string results = "";
            try {
                // 
                if (blogImage is not null) {
                    string altSizeList = blogImage.altSizeList;
                    Return_ImageDescription = blogImage.description;
                    Return_Imagename = blogImage.name;
                    Return_ThumbnailFilename = cp.Image.ResizeAndCrop(blogImage.Filename, app.blog.imageWidthMax, (int)Math.Round(app.blog.imageWidthMax * 0.75d), ref altSizeList, out bool isNewSize);
                    Return_ImageFilename = cp.Image.ResizeAndCrop(blogImage.Filename, app.blog.thumbnailImageWidth, 0, ref altSizeList, out bool isNewSize2);
                    if (isNewSize2 || isNewSize) {
                        blogImage.altSizeList = altSizeList;
                        blogImage.save(cp);
                    }
                }
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return results;
        }
        // 
        // ==================================================================================
        // 
        // 
    }
}